name: Tests

on:
  push:
    branches: [main]
  pull_request:
    branches: [main]
    types:
      - opened
      - ready_for_review
      - reopened
      - synchronize
  workflow_call:

jobs:
  test_package:
    name: Unit Tests
    runs-on: "ubuntu-20.04"
    steps:
      - uses: actions/checkout@v4
      - name: Setup python
        uses: actions/setup-python@v5
        with:
          python-version: "3.11"
          architecture: x64
          cache: pip
          cache-dependency-path: "**/*requirements*.txt"
      - name: Install dependencies
        run: |
          pip install -r code/requirements.txt -r code/dev-requirements.txt -r code/backend/requirements.txt
      - name: Run Python tests
        run: python -m pytest -m "not azure" --cov --cov-report xml:coverage.xml
        working-directory: code
      - uses: actions/upload-artifact@v4
        with:
          name: coverage
          path: code/coverage.xml
          if-no-files-found: error
